src/bigfile.h

```diff
@@ -100,15 +110,48 @@ int big_block_open(BigBlock * bb, const char * basename);
 int big_block_clear_checksum(BigBlock * bb);
 int big_block_create(BigBlock * bb, const char * basename, const char * dtype, int nmemb, int Nfile, const size_t fsize[]);
 int big_block_close(BigBlock * block);
+
+/** Initialise BigBlockPtr to the place in the BigBlock offset elements from the beginning of the block.
+ * This allows you to write into the BigBlock at a position other than the beginning.
+ * @param offset - Position to seek to in units of the size of the array element, eg, 8 bytes for an i8. If offset == -1, seek to the end of the array.
+ * All other offset < 0 seek array_size + 1 - offset.*/
```

Comment:
array_size + 1 - offset is a bug. If the code doesn't do (array_size + offset) then we need to fix it.